### ISSUE TITLE
BMD_cDFSRepGroup: DNSName vs ComputerName

### ISSUE THREAD

**Issue by nzspambot** [dsccommunity/DFSDsc#1]
Hi

First off, this is a wonderful module, so thanks for writing this

However I have some feedback, I've noticed in the resource BMD_cDFSRepGroup you compare on short name whereas my configs always have FQDNs for example:

```
 $ExistingMembers = @((Get-DfsrMember @Splat -ErrorAction Stop).ComputerName)

Compare-Object `
                -ReferenceObject $Members `
                -DifferenceObject $ExistingMembers
```

results in

```
InputObject            SideIndicator
-----------            -------------
DFS05                =>           
DFS06                =>           
DFS07                =>           
dfs05.fqdn.local <=           
dfs06.fqdn.local <=           
dfs07.fqdn.local <=  
```

which in my case _does not correctly test_

```
Replication Group "Templates".
VERBOSE: [DFS05]:                            [[cDFSRepGroup]WebTemplateGroup] Test-TargetResource: DFS Replication
Group "Templates" exists.
VERBOSE: [DFS05]:                            [[cDFSRepGroup]WebTemplateGroup] Test-TargetResource: DFS Replication
Group "Templates" members are different. Change required.  <--- THIS IS INCORRECT
VERBOSE: [DFS05]: LCM:  [ End    Test     ]  [[cDFSRepGroup]WebTemplateGroup]  in 1.6720 seconds.
VERBOSE: [DFS05]: LCM:  [ Start  Set      ]  [[cDFSRepGroup]WebTemplateGroup]
VERBOSE: [DFS05]:                            [[cDFSRepGroup]WebTemplateGroup] Set-TargetResource: Setting DFS
Replication Group "Templates".
VERBOSE: [DFS05]:                            [[cDFSRepGroup]WebTemplateGroup] Set-TargetResource: Ensuring DFS
Replication Group "Templates" exists.
VERBOSE: [DFS05]:                            [[cDFSRepGroup]WebTemplateGroup] Set-TargetResource: DFS Replication
Group "Templates" exists.
VERBOSE: [DFS05]: LCM:  [ End    Set      ]  [[cDFSRepGroup]WebTemplateGroup]  in 0.7810 seconds.
PowerShell DSC resource BMD_cDFSRepGroup  failed to execute Set-TargetResource functionality with error message: A
computer with the specified name already exists and cannot be added.
    + CategoryInfo          : InvalidOperation: (:) [], CimException
    + FullyQualifiedErrorId : ProviderOperationExecutionFailure
    + PSComputerName        : localhost

The SendConfigurationApply function did not succeed.
    + CategoryInfo          : NotSpecified: (root/Microsoft/...gurationManager:String) [], CimException
    + FullyQualifiedErrorId : MI RESULT 1
    + PSComputerName        : localhost

VERBOSE: Operation 'Invoke CimMethod' complete.
```

vs

```
 $ExistingMembers = @((Get-DfsrMember @Splat -ErrorAction Stop).DNSName)
```

which will _in my case_ does correctly test

I believe that the DNS name would be better as I have noticed that when I use shortnames the resources also bombs out

I am unsure however how this would work in a non domain environment as all the work is only ever done in a domain

thanks

**Comment by PlagueHO** [comment]
Thank you for the awesome comments :smile: good to know this resource gets some use!

That's a good catch - let me see if I can figure out a resolution to this. I'll see if I can get a fix out over the next day.

Thanks again for posting this issue too. :+1:

**Comment by PlagueHO** [comment]
And you're right - it looks like using the DNSName property of the DFSRMember object would resolve the problem. Will just need to check how this would affect any configs not using the computer name. I could code it to perform an FQDN/DNS check if the computername parameter of the config contained any "." otherwise perform a computername comparison.

I'll work on this tomorrow.

Thanks!

**Comment by nzspambot** [comment]
@PlagueHO awesome thanks for the quick reply, yeah I would deff test it out on a non domain setup if you can

Let me know if you need any help from my end

Also I have found another thing for you to look at ;) (new issue coming up)

thanks :)

**Comment by PlagueHO** [comment]
No problem!I  like to have resources in 100% working order so it's great to get feedback :+1:  I'm waiting on a code review of these resources by MSFT to try and get them moved over to the community DSC resources which will mean a lot more support and generally easier access to them. But it is taking a while to happen.

I think using these resources in a non-domain joined scenario should be fine because the unit and integration tests that run on commit always run against non-domain joined AppVeyor test rigs (can't easily domain-join them anyway).

Anyway, I'm hoping to have this released by the end of today (NZDT).

**Comment by PlagueHO** [comment]
I'm just merging the fix for this into the Dev branch now.

So once this is merged, feel free (in fact it would be much appreciated) to give this a try and let me know if you run into any problems.

In behind the scenes it now converts everything over the FQDNs (using the DomainName and ComputerName parameters if the ComputerName isn't already an FQDN) and matches the DNSName.

**Comment by nzspambot** [comment]
Yep cool; just looked at your code; looking good :+1:

**Comment by nzspambot** [comment]
@PlagueHO one question for you if I may; is there any reason why you don't change the version in your module manifest?

**Comment by PlagueHO** [comment]
Good question. Actually, I probably should - and will for this release.

I didn't used to because I have it set that the AppVeyor CI sets it when it creates and uploads the package to PowerShellGallery (and creates artifacts that I can download and use). Also, only AppVeyor can add the build number to the manifest as well. I never usually install the Module directly from Git so didn't run into an issue (I install it from the AppVeyor artifacts or the AppVeyor nuget package feed). I'll start setting it correctly :+1:

**Comment by nzspambot** [comment]
ah ok cool; yeah I have a CD pipeline which looks for changes on the module version which will push them to my pull server and internal nuget feed hence my question

cheers :+1:

**Comment by PlagueHO** [comment]
I think the way I'm doing it could definitely be better :smile: But given that I'm waiting for MSFT to accept it into the community resources I wasn't going to invest any time on it (they have their own internal release process etc).
